src/components/Reports/Shared/CalculationReports/DirectionButtons/DirectionButtons.tsx

```diff
@@ -1,30 +1,45 @@
 import { useState } from 'react';
 import { ChevronLeft, ChevronRight } from '@mui/icons-material';
 import { Box, Button } from '@mui/material';
-import { useFormikContext } from 'formik';
 import { useTranslation } from 'react-i18next';
-import { CalculationFormValues } from '../../Steps/StepThree/Calculation';
-import { SubmitModal } from '../../SubmitModal/SubmitModal';
-import { useMinisterHousingAllowance } from '../Context/MinisterHousingAllowanceContext';
+import { SubmitModal } from '../SubmitModal/SubmitModal';
 
 interface DirectionButtonsProps {
-  handleNext?: () => void;
-  isCalculate?: boolean;
+  handleNextStep?: () => void;
+  handlePreviousStep?: () => void;
+  buttonTitle?: string;
+  overrideNext?: () => void;
+  showBackButton?: boolean;
+
+  //Formik validation for submit modal
+  isSubmission?: boolean;
+  submitForm?: () => Promise<void>;
+  validateForm?: () => Promise<Record<string, string>>;
+  submitCount?: number;
+  isValid?: boolean;
 }
 
 export const DirectionButtons: React.FC<DirectionButtonsProps> = ({
-  handleNext,
-  isCalculate,
+  handleNextStep,
+  handlePreviousStep,
+  buttonTitle,
+  isSubmission,
+  overrideNext,
+  showBackButton,
+  submitForm,
+  validateForm,
+  submitCount,
+  isValid,
 }) => {
   const { t } = useTranslation();
 
-  const { handleNextStep, handlePreviousStep } = useMinisterHousingAllowance();
-
-  const { submitForm, validateForm, submitCount, isValid } =
-    useFormikContext<CalculationFormValues>();
   const [openConfirmation, setOpenConfirmation] = useState(false);
 
   const handleConfirm = async () => {
+    if (!submitForm || !validateForm) {
```

Comment:
This is for the `openConfirmation, setOpenConfirmation` state variable. Let's change this name to `openSubmitModal, setOpenSubmitModal`